cli/doc/parser.rs

```diff
@@ -48,7 +57,23 @@ impl From<BufferedError> for Vec<Diagnostic> {
   }
 }
 
+pub trait DocFileLoader {
+  fn resolve(
+    &self,
+    specifier: &str,
+    referrer: &str,
```

Comment:
Use stronger types - this is very much alike to `ModuleLoader` - but we don't care about compiled version, just TS